packages/core/react/src/WalletProviderBase.tsx

```diff
@@ -6,50 +6,48 @@ import type {
     WalletError,
     WalletName,
 } from '@solana/wallet-adapter-base';
+import React, { useCallback, useEffect, useMemo, useRef, useState } from 'react';
 import { WalletNotConnectedError, WalletNotReadyError, WalletReadyState } from '@solana/wallet-adapter-base';
+
 import type { PublicKey } from '@solana/web3.js';
-import type { FC, ReactNode } from 'react';
-import React, { useCallback, useEffect, useMemo, useRef, useState } from 'react';
-import { WalletNotSelectedError } from './errors.js';
-import { useLocalStorage } from './useLocalStorage.js';
-import type { Wallet } from './useWallet.js';
 import { WalletContext } from './useWallet.js';
+import { WalletNotSelectedError } from './errors.js';
+import type { WalletProviderProps } from './WalletProvider.js';
 
-export interface WalletProviderProps {
-    children: ReactNode;
-    wallets: Adapter[];
-    autoConnect?: boolean;
-    onError?: (error: WalletError) => void;
-    localStorageKey?: string;
-}
-
-const initialState: {
-    wallet: Wallet | null;
-    adapter: Adapter | null;
-    publicKey: PublicKey | null;
-    connected: boolean;
-} = {
-    wallet: null,
-    adapter: null,
-    publicKey: null,
-    connected: false,
-};
+type Props = Readonly<
+    Omit<WalletProviderProps, 'autoConnect' | 'localStorageKey'> & {
+        adapter: Adapter | null;
+        isUnloadingRef: React.RefObject<boolean>;
+        // NOTE: The presence/absence of this function implies that auto connect is enabled/disabled.
+        onAutoConnect?: () => Promise<void>;
+        onSelectWallet: (walletName: WalletName) => void;
+    }
+>;
 
-export const WalletProvider: FC<WalletProviderProps> = ({
+export function WalletProviderBase({
+    adapter,
```

Comment:
`WalletProviderBase` no longer assumes the responsibility of switching adapters. The higher-level `WalletProvider` does that. `WalletProviderBase` just takes the adapter it's given and prepares the `WalletContextState`.

Reply:
Cool, I like the approach. Once this is merged, we can just lightly change https://github.com/solana-labs/wallet-adapter/pull/528 to add the standard adapters to `WalletProvider`. If any maniacs want to use `WalletProviderBase`, they can get it unadulterated with sane defaults.